sdk/src/main/scala/com/horizen/SidechainHistory.scala

```diff
@@ -9,19 +9,22 @@ import com.horizen.node.util.MainchainBlockReferenceInfo
 import com.horizen.params.NetworkParams
 import com.horizen.storage.SidechainHistoryStorage
 import com.horizen.transaction.Transaction
-import com.horizen.utils.{WithdrawalEpochInfo, WithdrawalEpochUtils}
-import com.horizen.validation.SidechainBlockValidator
+import com.horizen.utils.{BytesUtils, WithdrawalEpochInfo, WithdrawalEpochUtils}
+import com.horizen.validation.{HistoryBlockValidator, SemanticBlockValidator}
 import scorex.core.NodeViewModifier
 import scorex.core.consensus.History._
 import scorex.core.consensus.{History, ModifierSemanticValidity}
 import scorex.core.validation.RecoverableModifierError
-import scorex.util.ModifierId
+import scorex.util.{ModifierId, idToBytes}
 
 import scala.compat.java8.OptionConverters._
 import scala.util.{Failure, Success, Try}
 
 
-class SidechainHistory private (val storage: SidechainHistoryStorage, params: NetworkParams, validators: Seq[SidechainBlockValidator])
+class SidechainHistory private (val storage: SidechainHistoryStorage,
+                                params: NetworkParams,
+                                semanticBlockValidators: Seq[SemanticBlockValidator],
+                                historyBlockValidators: Seq[HistoryBlockValidator])
```

Comment:
Proposition: checks in state also do via Validators (all validators have common trait).